tests/zquantum/qcbm/ansatz_test.py

```diff
@@ -166,31 +156,26 @@ def test_ansatz_circuit_four_layers(self, number_of_qubits, topology):
             np.ones(int((number_of_qubits * (number_of_qubits - 1)) / 2)),
         ]
 
-        expected_pycircuit = Program()
+        expected_first_layer = Circuit()
         for i in range(number_of_qubits):
-            expected_pycircuit += Program(pyquil.gates.RX(params[0][i], i))
+            expected_first_layer += RX(params[0][i])(i)
         for i in range(number_of_qubits):
-            expected_pycircuit += Program(
-                pyquil.gates.RZ(params[0][i + number_of_qubits], i)
-            )
-        expected_first_layer = Circuit(expected_pycircuit)
+            expected_first_layer += RZ(params[0][i + number_of_qubits])(i)
+
         expected_second_layer = get_entangling_layer(
-            params[1], number_of_qubits, "XX", topology
+            params[1], number_of_qubits, XX, topology
         )
-        expected_pycircuit = Program()
+
+        expected_third_layer = Circuit()
         for i in range(number_of_qubits):
-            expected_pycircuit += Program(pyquil.gates.RX(params[2][i], i))
+            expected_third_layer += RX(params[2][i])(i)
         for i in range(number_of_qubits):
-            expected_pycircuit += Program(
-                pyquil.gates.RZ(params[2][i + number_of_qubits], i)
-            )
+            expected_third_layer += RZ(params[2][i])(i)
         for i in range(number_of_qubits):
-            expected_pycircuit += Program(
-                pyquil.gates.RX(params[2][i + 2 * number_of_qubits], i)
-            )
-        expected_third_layer = Circuit(expected_pycircuit)
+            expected_third_layer += RX(params[2][i + number_of_qubits])(i)
+
```

Comment:
I think that the construction of the third layer differs from the original implementation. In particular, it seems to me that the second and third loops are different. Compare:
- `pyquil.gates.RZ(params[2][i + number_of_qubits], i)` vs. `expected_third_layer += RZ(params[2][i])(i)`
- `pyquil.gates.RX(params[2][i + 2 * number_of_qubits], i)` vs. `RX(params[2][i + number_of_qubits])(i)`

Reply:
Fixed in this and all the subsequent tests.
Also changed the params from `ones` to `random` to make these tests more reliable.
https://github.com/zapatacomputing/z-quantum-qcbm/pull/40/commits/c534e69c33b135d567032fc682b56cd0804331a8